### PR TITLE
Teams and custom fields

### PR DESCRIPTION
**Pull request by bob983**
### Description
- Added Custom Fields and Teams support (Pro Plan features)
  - there are new methods `getTeam` and `getTeams`
  - Custom Field responses were added to `Conversation` class
  - subclasses of `CustomFieldResponse` provide typed access to different custom field types. See [API documentation](http://developer.helpscout.net/help-desk-api/objects/field/)
- Added more tests for the ApiClient

### CODE REVIEW

**Comment by bob983** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;7NJ3Qy3A&#x2F;99-3-pro-plan-update-api-java-client-library"> (3) Pro Plan - Update API Java client library</a>

**Comment by bob983** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;7NJ3Qy3A&#x2F;99-3-pro-plan-update-api-java-client-library"> (3) Pro Plan - Update API Java client library</a>

File: src/main/java/net/helpscout/api/ApiClient.java [hunk 1]
```diff
@@ -877,12 +878,28 @@ public User getUser(Integer userID, List<String> fields) throws ApiException {
...
+     * Gets the user with the specified id.
+     *
+     * @param teamID
+     * @param fields
+     * @return User
+     * @throws ApiException
+     */
+    public Team getTeam(Long teamID, List<String> fields) throws ApiException {
+        if (teamID == null || teamID < 1) {
+            throw new ApiException("Invalid userId in getUser");
```
Comment (scalabilitysolved) [inline src/main/java/net/helpscout/api/ApiClient.java]: copy pasta of line 883? Perhaps that can be a little function we extract.

File: src/main/java/net/helpscout/api/model/Conversation.java [hunk 2]
```diff
@@ -21,7 +25,7 @@
     private boolean isDraft;
     private Long number;
     private Source source;
-    private UserRef owner;
+    private MailboxUserRef owner;
```
Comment (scalabilitysolved) [inline src/main/java/net/helpscout/api/model/Conversation.java:28]: Changing the type, is it going to impact any users of the client,  break serialization for anyone?

File: src/test/resources/responses/conversation_11.json [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        "openedAt": null,
+        "createdBy": {
+          "id": 150,
+          "firstName": "Karl",
+          "lastName": "Tester",
+          "email": "karl@worldoftesters.com",
+          "phone": "5552368",
+          "type": "customer",
+          "emails": [
+            "karl@worldoftesters.com",
```
Comment (scalabilitysolved) [inline src/test/resources/responses/conversation_11.json:68]: just checking this isn't a genuine email of one our customers is it?

**Comment by scalabilitysolved** [comment]
Couple of comments otherwise nice testing @bob983 :+1:

File: src/test/resources/responses/conversation_11.json (hunk 3 above)
Comment (bob983) [inline src/test/resources/responses/conversation_11.json:68]: It's my invention :) The domain does not exists, nor does the email.

File: src/main/java/net/helpscout/api/model/Conversation.java (hunk 2 above)
Comment (bob983) [inline src/main/java/net/helpscout/api/model/Conversation.java:28]: It might. Do you suggest to keep just `User` with type in the client?

Comment (bob983) [inline src/main/java/net/helpscout/api/model/Conversation.java:28]: OK, we're going to keep the new class. I updated the changelog.

**Comment by scalabilitysolved** [comment]
Change log looks good.

**Pull request merged**